### PR TITLE
[RM Ch07, RA1 Ch02] Update security Requirements

### LINKED ISSUE #1552: [RM Ch07, RA1 Ch02] Alignement with ONAP security requirements
Per discussion during CNTT-ONAP joint session, April's Virtual conference
References to ONAP securtity requirements, mainly about workloads can be added in chapter 6

### PR DESCRIPTION
**Pull request by karinesevilla**
Fixes  #1552 and #2031 
Applies to RM,RA1 and RA2
Following the review of ONAP requirements collected in https://wiki.lfnetworking.org/display/LN/RM+and+RA-1%3A+ONAP+Security+Requirements, relevant requirements reviewed during RA#1 meeting, October 19th.
It concerns:
sec.sys.015, sec.sys.016, sec.lcm.002, sec.lcm.012, sec.mon.001, sec.mon.018, sec.mon.019, sec.mon.020
It applies also to RA2, issue#2009 updated

### CODE REVIEW

**Review by walterkozlowski — APPROVED** [review]

**Comment by karinesevilla** [comment]
Ready to be reviewed
For details, check the table  https://wiki.lfnetworking.org/display/LN/RM+and+RA-1%3A+ONAP+Security+Requirements

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 1]
```diff
@@ -227,6 +227,12 @@ These rows are removed and commented out as it's not clear what the requirement
...
 | sec.sys.012 | Access | The Platform **must** protect all secrets by using strong encryption techniques and storing the protected secrets externally from the component (e.g., in OpenStack Barbican) | [RA-1 6.3.1 "Platform Access"](./chapter06.md#631-platform-access) |
 | sec.sys.013 | Access | The Platform **must** generate secrets dynamically as and when needed. | [RA-1 6.3.1 "Platform Access"](./chapter06.md#631-platform-access) |
+| sec.sys.015 | Access | The platform **must not** contain back door entries (unpublished access points, APIs, etc.). |  |
+| sec.sys.016 | Access | Login access to the platform's components **must** be through encrypted protocols such as SSH v2 or TLS v1.2 or higher. | Note: Hardened jump servers isolated from external networks are recommended |
+| sec.sys.017 | Access | The platform must provide the capability of using digital certificates that comply with X.509 standards and issued from a trusted Certification Authority. |  |
+| sec.sys.018 | Access | The paltform must provide the capability of allowing certificate renewal and revocation. |  |
```
Comment (petorre) [inline doc/ref_arch/openstack/chapters/chapter02.md:233]: paltform -> platform

**Review by petorre — APPROVED** [review]

File: doc/ref_arch/openstack/chapters/chapter02.md (hunk 1 above)
Comment (karinesevilla) [reply doc/ref_arch/openstack/chapters/chapter02.md]: @petorre Thanks Petar!
Done

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 2]
```diff
@@ -227,6 +227,12 @@ These rows are removed and commented out as it's not clear what the requirement
...
 | sec.sys.012 | Access | The Platform **must** protect all secrets by using strong encryption techniques and storing the protected secrets externally from the component (e.g., in OpenStack Barbican) | [RA-1 6.3.1 "Platform Access"](./chapter06.md#631-platform-access) |
 | sec.sys.013 | Access | The Platform **must** generate secrets dynamically as and when needed. | [RA-1 6.3.1 "Platform Access"](./chapter06.md#631-platform-access) |
+| sec.sys.015 | Access | The platform **must not** contain back door entries (unpublished access points, APIs, etc.). |  |
+| sec.sys.016 | Access | Login access to the platform's components **must** be through encrypted protocols such as SSH v2 or TLS v1.2 or higher. | Note: Hardened jump servers isolated from external networks are recommended |
+| sec.sys.017 | Access | The platform must provide the capability of using digital certificates that comply with X.509 standards and issued from a trusted Certification Authority. |  |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:232]:
```suggestion
| sec.sys.017 | Access | The platform **must** provide the capability of using digital certificates that comply with X.509 standards and issued from a trusted Certification Authority. |  |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 3]
```diff
@@ -227,6 +227,12 @@ These rows are removed and commented out as it's not clear what the requirement
...
+| sec.sys.018 | Access | The platform must provide the capability of allowing certificate renewal and revocation. |  |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:233]:
```suggestion
| sec.sys.018 | Access | The platform **must** provide the capability of allowing certificate renewal and revocation. |  |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 4]
```diff
@@ -227,6 +227,12 @@ These rows are removed and commented out as it's not clear what the requirement
...
+| sec.sys.019 | Access | The platform must provide the capability of testing the validity of a digital certificate (CA signature, validity period, non revocation, identity). |  |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:234]:
```suggestion
| sec.sys.019 | Access | The platform **must** provide the capability of testing the validity of a digital certificate (CA signature, validity period, non revocation, identity). |  |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 5]
```diff
@@ -272,7 +278,7 @@ These rows are removed and commented out as it's not clear what the requirement
 | Ref # | sub-category | Description |  Traceability |
 |---|----|---|----|
 | sec.lcm.001 | LCM | The Platform **must** support Secure Provisioning, Availability, and Deprovisioning (Secure Clean-Up) of workload resources where Secure Clean-Up includes tear-down, defense against virus or other attacks. | [RA-1 6.3.7 Security Audit Logging](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter06.md#637-security-audit-logging) |  
-| sec.lcm.002 | LCM | Operational **must** use management protocols limiting security risk such as SNMPv3, SSH v2, ICMP, NTP, syslog and TLS v1.2 or higher. | [RA-1 6.3.7 Security Audit Logging](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter06.md#637-security-audit-logging) |
+| sec.lcm.002 | LCM | Cloud operations staff and systems **must** use management protocols limiting security risk such as SNMPv3, SSH v2, ICMP, NTP, syslog and TLS v1.2 or higher. | [RA-1 6.3.7 Security Audit Logging](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter06.md#637-security-audit-logging) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:281]:
```suggestion
| sec.lcm.002 | LCM | The Cloud Operator **must** use management protocols limiting security risk such as SNMPv3, SSH v2, ICMP, NTP, syslog and TLS v1.2 or higher. | [RA-1 6.3.7 Security Audit Logging](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter06.md#637-security-audit-logging) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 6]
```diff
@@ -281,6 +287,8 @@ These rows are removed and commented out as it's not clear what the requirement
 | sec.lcm.009 | LCM | The Platform **must** be able to update the tag of newly instantiated, suspended, hibernated, migrated and restarted images with relevant geolocation (geographical) information. | [RA-1 6.3.7 Security Audit Logging](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter06.md#637-security-audit-logging) |
 | sec.lcm.010 | LCM | The Platform **must** log all changes to geolocation along with the mechanisms and sources of location information (i.e. GPS, IP block, and timing). | [RA-1 6.3.7 Security Audit Logging](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter06.md#637-security-audit-logging) |
 | sec.lcm.011 | LCM | The Platform **must** implement Security life cycle management processes including the proactive update and patching of all deployed Cloud Infrastructure software. | [RA-1 6.3.7 Security Audit Logging](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/openstack/chapters/chapter06.md#637-security-audit-logging) |
+| sec.lcm.012 | LCM | The platform **must** log any privilege escalation. |  |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:290]:
```suggestion
| sec.lcm.012 | LCM | The platform **must** log any access privilege escalation. |  |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 7]
```diff
@@ -302,8 +310,12 @@ The Platform is assumed to provide configurable alerting and notification capabi
 | sec.mon.011 | Monitoring/Audit | The Platform **must** Monitor and Audit logs from infrastructure elements and workloads to detected anomalies in the system components and take corrective actions accordingly. | [RA-1 6.3.7.1 "Creating logs"](./chapter06.md#6371-creating-logs) |
 | sec.mon.012 | Monitoring/Audit | The Platform **must** Monitor and Audit Traffic patterns and volumes to prevent malware download attempts. |  |
 | sec.mon.013 | Monitoring | The monitoring system **must not** affect the security (integrity and confidentiality) of the infrastructure, workloads, or the user data (through back door entries). |  |
-| sec.mon.015 | Monitoring | The Platform **must** ensure that the Monitoring systems are never starved of resources. |  |
-| sec.lcm.017 | Audit | The Platform **must** Audit systems for any missing security patches and take appropriate actions. | [RA-1 6.3.2.3 "Patches"](./chapter06.md#6323-patches) |
+| sec.mon.015 | Monitoring | The Platform **must** ensure that the Monitoring systems are never starved of resources and **must** activate alarms when resources exceeded a configurable threshold. |  |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:313]:
```suggestion
| sec.mon.015 | Monitoring | The Platform **must** ensure that the Monitoring systems are never starved of resources and **must** activate alarms when resource utilisation exceeds a configurable threshold. |  |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 8]
```diff
@@ -302,8 +310,12 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
+| sec.mon.017 | Audit | The Platform **must** Audit systems for any missing security patches and take appropriate actions. | [RA-1 6.3.2.3 "Patches"](./chapter06.md#6323-patches) |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:314]:
```suggestion
| sec.mon.017 | Audit | The Platform **must** audit systems for any missing security patches and take appropriate actions. | [RA-1 6.3.2.3 "Patches"](./chapter06.md#6323-patches) |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 9]
```diff
@@ -302,8 +310,12 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
+| sec.mon.018 | Monitoring | The platform, starting from initialization, **must** collect and analyze logs to identify security events, and store these events in an external system. | |
+| sec.mon.019 | Monitoring | The platform’s components **must not** include an authentication credential, e.g., password, in any logs, even if encrypted. | | 
+| sec.mon.020 | Monitoring/Audit | The platform’s logging system **must** support the storage of security audit logs for a configurable period of time. | |
+| sec.mon.021 | Monitoring | The platform **must** store security events locally if the external logging system is unavailable and shall attempt to send these to the logging system when communications are re-established. | |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:318]:
```suggestion
| sec.mon.021 | Monitoring | The platform **must** store security events locally if the external logging system is unavailable and shall attempt to send these to the external logging system when communications are re-established. | |
```

File: doc/ref_model/chapters/chapter07.md [hunk 10]
```diff
@@ -427,6 +427,12 @@ Security certification should encompass the following elements:
 | req.sec.sys.012 | The Platform **must** protect all secrets by using strong encryption techniques, and storing the protected secrets externally from the component. | (e.g., in OpenStack Barbican). |
 | req.sec.sys.013 | The Platform **must** provide secrets dynamically as and when needed. | |
 | req.sec.sys.014 | The Platform **should** use Linux Security Modules such as SELinux to control access to resources. | |
+| req.sec.sys.015 | The platform **must not** contain back door entries (unpublished access points, APIs, etc.). |  |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:430]:
```suggestion
| req.sec.sys.015 | The Platform **must not** contain back door entries (unpublished access points, APIs, etc.). |  |
```

File: doc/ref_model/chapters/chapter07.md [hunk 11]
```diff
@@ -427,6 +427,12 @@ Security certification should encompass the following elements:
...
+| req.sec.sys.016 | Login access to the platform's components **must** be through encrypted protocols such as SSH v2 or TLS v1.2 or higher. | Note: Hardened jump servers isolated from external networks are recommended |
+| req.sec.sys.017 | The platform must provide the capability of using digital certificates that comply with X.509 standards and issued from a trusted Certification Authority. |  |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:432]:
```suggestion
| req.sec.sys.017 | The Platform must provide the capability of using digital certificates that comply with X.509 standards issued by a trusted Certification Authority. |  |
```

File: doc/ref_model/chapters/chapter07.md [hunk 12]
```diff
@@ -427,6 +427,12 @@ Security certification should encompass the following elements:
...
+| req.sec.sys.018 | The platform must provide the capability of allowing certificate renewal and revocation. |  |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:433]:
```suggestion
| req.sec.sys.018 | The Platform must provide the capability of allowing certificate renewal and revocation. |  |
```

File: doc/ref_model/chapters/chapter07.md [hunk 13]
```diff
@@ -427,6 +427,12 @@ Security certification should encompass the following elements:
...
+| req.sec.sys.019 | The platform must provide the capability of testing the validity of a digital certificate (CA signature, validity period, non revocation, identity). |  |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:434]:
```suggestion
| req.sec.sys.019 | The Platform must provide the capability of testing the validity of a digital certificate (CA signature, validity period, non revocation, identity). |  |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 14]
```diff
@@ -227,6 +227,12 @@ These rows are removed and commented out as it's not clear what the requirement
 | sec.sys.011 | Access | The Platform **must not** reuse the same authentication credentials (e.g., key pairs) on different Platform components (e.g., different hosts, or different services). | [RA-1 6.3.1 "Platform Access"](./chapter06.md#631-platform-access) |
 | sec.sys.012 | Access | The Platform **must** protect all secrets by using strong encryption techniques and storing the protected secrets externally from the component (e.g., in OpenStack Barbican) | [RA-1 6.3.1 "Platform Access"](./chapter06.md#631-platform-access) |
 | sec.sys.013 | Access | The Platform **must** generate secrets dynamically as and when needed. | [RA-1 6.3.1 "Platform Access"](./chapter06.md#631-platform-access) |
+| sec.sys.015 | Access | The platform **must not** contain back door entries (unpublished access points, APIs, etc.). |  |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:230]:
```suggestion
| sec.sys.015 | Access | The Platform **must not** contain back door entries (unpublished access points, APIs, etc.). |  |
```

File: doc/ref_arch/openstack/chapters/chapter02.md [hunk 15]
```diff
@@ -227,6 +227,12 @@ These rows are removed and commented out as it's not clear what the requirement
...
+| sec.sys.016 | Access | Login access to the platform's components **must** be through encrypted protocols such as SSH v2 or TLS v1.2 or higher. | Note: Hardened jump servers isolated from external networks are recommended |
```
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:231]:
```suggestion
| sec.sys.016 | Access | Login access to the Platform's components **must** be through encrypted protocols such as SSH v2 or TLS v1.2 or higher. | Note: Hardened jump servers isolated from external networks are recommended |
```

File: doc/ref_arch/openstack/chapters/chapter02.md (hunk 2 above)
Comment (pgoyal01) [inline doc/ref_arch/openstack/chapters/chapter02.md:232]:
```suggestion
| sec.sys.017 | Access | The Platform must provide the capability of using digital certificates that comply with X.509 standards issued by a trusted Certification Authority. |  |
```

File: doc/ref_model/chapters/chapter07.md [hunk 16]
```diff
@@ -516,9 +523,13 @@ The Platform is assumed to provide configurable alerting and notification capabi
 | req.sec.mon.012 | The Platform **must** Monitor and Audit Traffic patterns and volumes to prevent malware download attempts. | |
 | req.sec.mon.013 | The monitoring system **must not** affect the security (integrity and confidentiality) of the infrastructure, workloads, or the user data (through back door entries). |  |
 | req.sec.mon.014 | The Monitoring systems **should not** impact IAAS, PAAS, and SAAS SLAs including availability SLAs. |  |
-| req.sec.mon.015 | The Platform **must** ensure that the Monitoring systems are never starved of resources. |  |
+| req.sec.mon.015 | The Platform **must** ensure that the Monitoring systems are never starved of resources and **must** activate alarms when resources exceeded a configurable threshold. |  |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:526]:
```suggestion
| req.sec.mon.015 | The Platform **must** ensure that the Monitoring systems are never starved of resources and **must** activate alarms when resource utilisation exceeds a configurable threshold. |  |
```

File: doc/ref_model/chapters/chapter07.md [hunk 17]
```diff
@@ -516,9 +523,13 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
 | req.sec.mon.016 | The Platform Monitoring components **should** follow security best practices for auditing, including secure logging and tracing. | |
-| req.sec.lcm.017 | The Platform **must** audit systems for any missing security patches and take appropriate actions. |  |
+| req.sec.mon.017 | The Platform **must** audit systems for any missing security patches and take appropriate actions. |  |
+| req.sec.mon.018 | The platform, starting from initialization, **must** collect and analyze logs to identify security events, and store these events in an external system. | |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:529]:
```suggestion
| req.sec.mon.018 | The Platform, starting from initialization, **must** collect and analyze logs to identify security events, and store these events in an external system. | |
```

File: doc/ref_model/chapters/chapter07.md [hunk 18]
```diff
@@ -516,9 +523,13 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
+| req.sec.mon.019 | The platform’s components **must not** include an authentication credential, e.g., password, in any logs, even if encrypted. | |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:530]:
```suggestion
| req.sec.mon.019 | The Platform’s components **must not** include an authentication credential, e.g., password, in any logs, even if encrypted. | | 
```

File: doc/ref_model/chapters/chapter07.md [hunk 19]
```diff
@@ -516,9 +523,13 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
-| req.sec.mon.015 | The Platform **must** ensure that the Monitoring systems are never starved of resources. |  |
+| req.sec.mon.015 | The Platform **must** ensure that the Monitoring systems are never starved of resources and **must** activate alarms when resources exceeded a configurable threshold. |  |
 | req.sec.mon.016 | The Platform Monitoring components **should** follow security best practices for auditing, including secure logging and tracing. | |
-| req.sec.lcm.017 | The Platform **must** audit systems for any missing security patches and take appropriate actions. |  |
+| req.sec.mon.017 | The Platform **must** audit systems for any missing security patches and take appropriate actions. |  |
+| req.sec.mon.018 | The platform, starting from initialization, **must** collect and analyze logs to identify security events, and store these events in an external system. | |
+| req.sec.mon.019 | The platform’s components **must not** include an authentication credential, e.g., password, in any logs, even if encrypted. | | 
+| req.sec.mon.020 | The platform’s logging system **must** support the storage of security audit logs for a configurable period of time. | |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:531]:
```suggestion
| req.sec.mon.020 | The Platform’s logging system **must** support the storage of security audit logs for a configurable period of time. | |
```

File: doc/ref_model/chapters/chapter07.md [hunk 20]
```diff
@@ -516,9 +523,13 @@ The Platform is assumed to provide configurable alerting and notification capabi
...
+| req.sec.mon.021 | The platform **must** store security events locally if the external logging system is unavailable and shall attempt to send these to the logging system when communications are re-established. | |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.md:532]:
```suggestion
| req.sec.mon.021 | The platform **must** store security events locally if the external logging system is unavailable and shall periodically attempt to send these to the external logging system until successful.. | |
```

**Comment by walterkozlowski** [comment]
@pgoyal01, I have committed your suggestions.  Please approve and I will merge.

**Review by pgoyal01 — APPROVED** [review]

**Pull request merged**